File: Leetcode/LongestCommonPrefix.cs
// Write a function to find the longest common prefix string amongst an array of strings.
// If there is no common prefix, return an empty string "".

public class Solution
{
    public string LongestCommonPrefix(string[] strs)
    {
        if (strs.Length == 0)
            return "";
        if (strs.Length == 1)
            return strs[0];

        var prefix = strs[0];
        var pLength = prefix.Length;

        for (int i = 1; i < strs.Length; i++)
            while (strs[i].IndexOf(prefix) != 0)
            {
                prefix = prefix.Substring(0, pLength - 1);
                pLength--;
                if (pLength == 0)
                    return "";
            }

        return prefix;
    }
}

File: Leetcode/LongestCommonPrefix.py
# Write a function to find the longest common prefix string amongst an array of strings.
# If there is no common prefix, return an empty string "".

def longestCommonPrefix(self, strs):
        if len(strs) == 0:
            return ""
        if len(strs) == 1:
            return strs[0]
        
        prefix = strs[0]
        plen = len(prefix)
        
        for string in strs[1:]:
            while prefix != string[:plen]:
                prefix = prefix[:plen - 1]
                plen -= 1
                if plen == 0:
                    return ""
        
        return prefix